src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "wall_banner")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "bedrock")),
+            ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "portal")))
+    var material: Block = ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "obsidian"))!!
+    var fillerMat: Block = ForgeRegistries.BLOCKS.getValue(ResourceLocation("minecraft", "netherrack"))!!
+    private var playerHotbarSlot = -1
+    private var lastHotbarSlot = -1
+    private var isSneaking = false
+    var pathing = true
+    private var stuckDetector = 0
+    private var buildDirectionSaved = 0
+    private var buildDirectionCoordinate = 0
+    private val directions = listOf("North", "North-East", "East", "South-East", "South", "South-West", "West", "North-West")
+
+    //Stats
+    private var totalBlocksPlaced = 0
+    var totalBlocksDestroyed = 0
+    private var totalBlocksDistance = 0
+
+    //val blockQueue = PriorityQueue<BlockTask>(TaskComparator())
+    var blockQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private val doneQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private var blockOffsets = mutableListOf<Pair<BlockPos, Boolean>>()
+    private var waitTicks = 0
+    private var blocksPlaced = 0
+    private lateinit var currentBlockPos: BlockPos
+    private lateinit var startingBlockPos: BlockPos
+
+    fun isDone(): Boolean { return blockQueue.size == 0 }
+
+    override fun onEnable() {
+        if (mc.player == null) {
+            disable()
+            return
+        }
+        buildDirectionSaved = getPlayerDirection()
+        startingBlockPos = BlockPos(floor(mc.player.posX).toInt(), floor(mc.player.posY).toInt(), floor(mc.player.posZ).toInt())
+        currentBlockPos = startingBlockPos
+        playerHotbarSlot = mc.player.inventory.currentItem
+        lastHotbarSlot = -1
+
+        playerHotbarSlot = mc.player.inventory.currentItem
+        buildDirectionCoordinate = if (buildDirectionSaved == 0 || buildDirectionSaved == 4) { startingBlockPos.getX() }
+        else if (buildDirectionSaved == 2 || buildDirectionSaved == 6) { startingBlockPos.getZ() }
+        else { 0 }
+
+        blockQueue.clear()
+        doneQueue.clear()
+        updateTasks()
+        printEnable()
+    }
+
+    override fun onDisable() {
+        if (mc.player == null) return
+
+        // load initial player hand
+        if (lastHotbarSlot != playerHotbarSlot && playerHotbarSlot != -1) {
+            mc.player.inventory.currentItem = playerHotbarSlot
+        }
+        if (isSneaking) {
+            mc.player.connection.sendPacket(CPacketEntityAction(mc.player, CPacketEntityAction.Action.STOP_SNEAKING))
+            isSneaking = false
+        }
+        playerHotbarSlot = -1
+        lastHotbarSlot = -1
+
+        BaritoneAPI.getProvider().primaryBaritone.pathingBehavior.cancelEverything()
```

Comment:
is this really needed? Can't we get the HighwayToolProcess and cancel only that one?